src/Module.zig

```diff
@@ -1953,7 +1959,15 @@ pub const EmbedFile = struct {
 pub const ErrorMsg = struct {
     src_loc: SrcLoc,
     msg: []const u8,
-    notes: []ErrorMsg = &.{},
+    notes: []Note = &.{},
```

Comment:
The intent here is that notes can have sub-notes. Are you interested in making a case against this?

Reply:
I just didn't find any examples of notes having notes attached, how would that differ from just having them all linearly?

Reply:
One example: the referenced by trace could be implemented in this way, having one of the notes be the root referenced by note, and then having each one attached to that one. However I'm not against explicitly handling it with a different data type as you do here.

For terminals, the tree could be represented with indentation. For IDEs it could be represented as a tree view widget. We do have *some* prior art for tree view on compile errors in the case that compiler-rt has compile errors.